ostc4pack/src/OSTC4pack_V4.cpp

```
@@ -436,9 +451,40 @@ int main(int argc, char** argv) {
 	}
 	len = fread(buf, sizeof(char), sizeof(buf), fp);
 	printf("%d bytes read (hex: %#x )\n", (uint32_t)len, (uint32_t)len);
+
+    if (type == 3) {
+		if (len > 0x01fff0) {
+			printf("Error: File too large for type 3.\n");
+			return -1;
```

Cppcheck (2.19.0)

[error] Resource leak

(resourceLeak)